cou/steps/analyze.py

```diff
@@ -78,6 +79,30 @@ def _split_control_plane_and_data_plane(
                 control_plane.append(app)
         return control_plane, data_plane
 
+    @classmethod
+    def _remove_data_plane_apps(
+        cls, control_plane: list[OpenStackApplication], data_plane: list[OpenStackApplication]
+    ) -> None:
+        """Move control plane applications to data plane if any of its units is on data plane.
+
+        :param control_plane: List of control plane applications.
+        :type control_plane:  list[OpenStackApplication]
+        :param data_plane: List of data plane applications.
+        :type data_plane:  list[OpenStackApplication]
+        """
+        data_plane_machines = set().union(
+            *[{unit.machine for unit in app.units} for app in data_plane]
+        )
+
+        apps_to_move = []
+        for app in control_plane:
+            if len([unit for unit in app.units if unit.machine in data_plane_machines]) > 0:
+                apps_to_move.append(app)
+
+        for app in apps_to_move:
+            control_plane.remove(app)
+            data_plane.append(app)
```

Comment:
This is not good design pattern to update input arguments, this function should instead return two lists.

Reply:
done